### PR TITLE
Add support for User Flows

### PR DESCRIPTION
**Pull request by JasonPapanicholasWF**
## Description

Please provide a meaningful description of what this change will do, or is for. Bonus points for including links to related issues, other PRs, or technical references.

Note that by _not_ including a description, you are asking reviewers to do extra work to understand the context of this change, which may lead to your PR taking much longer to review, or result in it not being reviewed at all.

## Type of Change

- [ ] Bug Fix
- [ ] New Feature
- [ ] Breaking Change
- [ ] Refactor
- [ ] Documentation
- [ ] Other (please describe)

## Checklist

- [ ] I have read the [contributing guidelines](https://github.com/wayfair-incubator/oss-template/blob/main/CONTRIBUTING.md)
- [ ] Existing issues have been referenced (where applicable)
- [ ] I have verified this change is not present in other open pull requests
- [ ] Functionality is documented
- [ ] All code style checks pass
- [ ] New code contribution is covered by automated tests
- [ ] All new and existing tests pass

### CODE REVIEW

File: flows/my-test.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
+const api = require('lighthouse/lighthouse-core/fraggle-rock/api.js');
+
+class whateverYouWant {
```
Comment (spember) [inline flows/my-test.js]: maybe a more meaningful name for the test file? ;)

File: packages/race-cli/src/scenarios/racer-client.ts [hunk 2]
```diff
@@ -60,12 +61,36 @@ export const handleStartRacer = async ({
...
+    .catch((error: AxiosError) => handleRacerError(error));
+
 /*
   Validate the response is either HTML or a LHR
 */
-const validateResponseData = (data: LighthouseResults | string) => {
+const validateResponseData = (data: any | string) => {
   if (
     (typeof data !== 'string' && data.lighthouseVersion) ||
+    // TEMP
```
Comment (spember) [inline packages/race-cli/src/scenarios/racer-client.ts]: temp? temp what?

File: packages/racer/src/profiling/config.ts [hunk 3]
```diff
@@ -14,6 +23,51 @@ export interface RaceProfileCommand {
...
+
+  constructor(
+    requestedJobId: number,
+    command: RaceFlowCommand,
+    testCase: TestCaseType
+  ) {
+    this.jobId = requestedJobId;
+    this.testCase = testCase;
+    this.deviceType = command.deviceType || this.deviceType;
+    // this.disableStorageReset =
```
Comment (spember) [inline packages/racer/src/profiling/config.ts:59]: do we still need this commented out code?

File: packages/racer/src/controllers/results.ts [hunk 4]
```diff
@@ -61,7 +61,7 @@ const selectResponseTypeByHeader = (
       MIME_HTML
     );
   } else {
-    return new EndpointResponse(record.results.lhr).withHeader(
+    return new EndpointResponse(record.results.steps).withHeader(
```
Comment (spember) [inline packages/racer/src/controllers/results.ts:64]: did you end up re-writing the normal `profile` command to return a step of 1 here?

File: packages/racer/src/profiling/index.ts [hunk 5]
```diff
@@ -40,9 +95,22 @@ const profileWithLighthouse = async (context: RaceContext) => {
     lighthouseFlags,
     context.blockedUrlPatterns
   );
+
+  // Re-format the results object into a shared user flow type
+  // Essentially a flow with a single step
+  const formattedResults: UserFlowResultsWrapper = {
```
Comment (spember) [inline packages/racer/src/profiling/index.ts:101]: ah, here it is. nice

**Review by spember — APPROVED** [review]
lgtm to me, nice work. 

That being said, is it too early to update the Readme.md to describe how to use the flow?

File: packages/racer/src/profiling/config.ts (hunk 3 above)
Comment (JasonPapanicholasWF) [reply packages/racer/src/profiling/config.ts:59]: I think we'll eventually want to wire these in

File: packages/racer/src/controllers/results.ts (hunk 4 above)
Comment (JasonPapanicholasWF) [reply packages/racer/src/controllers/results.ts:64]: Yes both commands have been tested to work

**Pull request merged**